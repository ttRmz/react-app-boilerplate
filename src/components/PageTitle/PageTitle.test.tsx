import { fireEvent, render } from '@testing-library/react'
import { PageTitle } from '.'

describe('<PageTitle />', () => {
  it('should render a PageTitle', () => {
    const { container } = render(<PageTitle children="test title" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should pass props on root element', () => {
    const onClick = jest.fn()
    const testId = 'title-id'

    const { container, getByTestId } = render(
      <PageTitle
        children="test title"
        className="test-class"
        id="test-id"
        onClick={onClick}
        data-testid={testId}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(getByTestId(testId))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
