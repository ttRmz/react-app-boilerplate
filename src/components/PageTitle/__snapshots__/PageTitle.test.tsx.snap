// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> should pass props on root element 1`] = `
<h1
  class="test-class"
  data-testid="title-id"
  id="test-id"
>
  test title
</h1>
`;

exports[`<PageTitle /> should render a PageTitle 1`] = `
<h1
  class="PageTitle"
>
  test title
</h1>
`;

exports[`Storyshots UI/Asset Demo 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "inline-flex",
      "flexDirection": "column",
    }
  }
>
  <figure
    className="Asset"
    style={
      Object {
        "fontSize": "48px",
        "marginBottom": "12px",
      }
    }
  >
    <img
      alt="analysis"
      className="Asset__image"
      src="http://localhost/assets/symbols/analysis.svg"
    />
  </figure>
  <span>
    analysis
  </span>
</div>
`;

exports[`Storyshots UI/Asset List 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexFlow": "row wrap",
      "justifyContent": "center",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 36,
        "margin": 12,
        "width": "4em",
      }
    }
  >
    <figure
      className="Asset"
    >
      <img
        alt="analysis"
        className="Asset__image"
        src="http://localhost/assets/symbols/analysis.svg"
      />
    </figure>
    <span
      style={
        Object {
          "fontSize": 12,
          "marginTop": 14,
          "textAlign": "center",
        }
      }
    >
      analysis
    </span>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 36,
        "margin": 12,
        "width": "4em",
      }
    }
  >
    <figure
      className="Asset"
    >
      <img
        alt="analytics"
        className="Asset__image"
        src="http://localhost/assets/symbols/analytics.svg"
      />
    </figure>
    <span
      style={
        Object {
          "fontSize": 12,
          "marginTop": 14,
          "textAlign": "center",
        }
      }
    >
      analytics
    </span>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 36,
        "margin": 12,
        "width": "4em",
      }
    }
  >
    <figure
      className="Asset"
    >
      <img
        alt="newspaper"
        className="Asset__image"
        src="http://localhost/assets/symbols/newspaper.svg"
      />
    </figure>
    <span
      style={
        Object {
          "fontSize": 12,
          "marginTop": 14,
          "textAlign": "center",
        }
      }
    >
      newspaper
    </span>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 36,
        "margin": 12,
        "width": "4em",
      }
    }
  >
    <figure
      className="Asset"
    >
      <img
        alt="report"
        className="Asset__image"
        src="http://localhost/assets/symbols/report.svg"
      />
    </figure>
    <span
      style={
        Object {
          "fontSize": 12,
          "marginTop": 14,
          "textAlign": "center",
        }
      }
    >
      report
    </span>
  </div>
</div>
`;

exports[`Storyshots UI/Icon Demo 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "inline-flex",
      "flexDirection": "column",
    }
  }
>
  <i
    className="Icon Icon-react"
    style={
      Object {
        "fontSize": "48px",
        "marginBottom": "12px",
      }
    }
  />
  <span>
    react
  </span>
</div>
`;

exports[`Storyshots UI/Icon List 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexFlow": "row wrap",
      "justifyContent": "center",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 36,
        "margin": 12,
        "width": "4em",
      }
    }
  >
    <i
      className="Icon Icon-heart"
    />
    <span
      style={
        Object {
          "fontSize": 12,
          "marginTop": 14,
          "textAlign": "center",
        }
      }
    >
      heart
    </span>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 36,
        "margin": 12,
        "width": "4em",
      }
    }
  >
    <i
      className="Icon Icon-react"
    />
    <span
      style={
        Object {
          "fontSize": 12,
          "marginTop": 14,
          "textAlign": "center",
        }
      }
    >
      react
    </span>
  </div>
</div>
`;

exports[`Storyshots UI/PageTitle Demo 1`] = `
<h1
  className="PageTitle"
>
  Hello world
</h1>
`;
